base/range.jl

```diff
@@ -86,51 +77,140 @@ julia> range(1, 10, length=101)
 
 julia> range(1, 100, step=5)
 1:5:96
+
+julia> range(stop=10, length=5)
+6:10
+
+julia> range(stop=10, step=1, length=5)
+6:1:10
+
+julia> range(start=1, step=1, stop=10)
+1:1:10
 ```
+If `length` is not specified and `stop - start` is not an integer multiple of `step`, a range that ends before `stop` will be produced.
+```jldoctest
+julia> range(1, 3.5, step=2)
+1.0:2.0:3.0
+```
+
+Special care is taken to ensure intermediate values are computed rationally.
+To avoid this induced overhead, see the [`LinRange`](@ref) constructor.
+
+`start` may be specified as either a positional or keyword argument.
+`stop` may be specified as either a positional or keyword argument. If it is specified as a positional argument,
+one of `step` or `length` must also be provided.
```

Comment:
```suggestion
Both `start`  and `stop` may be specified as either a positional or keyword arguments.
If both are specified as positional arguments, one of `step` or `length` must also be provided.
```

Boy this is a funny special case.  I understand why you did it this way (backwards compatibility) but we may want to flag it for changing in the future.